### ISSUE TITLE
Handle color space transformation correctly for JPEG image compressed in RGB color space

### ISSUE THREAD

**Issue by hackermd** [python-pillow/Pillow#3952]
Pillow currently doesn't correctly read JPEG compressed images, which were compressed in RGB color space, i.e. which were not transformed into YCbCr color space upon compression.

**Comment by hackermd** [comment]
The fundamental issue is that the `JpegImagePlugin` assumes that images have been transformed into YCbCr color space upon compression. This would be a valid assumption for JFIF (JPEG File Interchange Format), which requires compression in YCbCr color space.

However, the documentation of the plugin is ambiguous about whether it implements JFIF or JPEG. The comment at the top of the module states:

```None
# JPEG (JFIF) file handling
#
# See "Digital Compression and Coding of Continuous-Tone Still Images,
# Part 1, Requirements and Guidelines" (CCITT T.81 / ISO 10918-1)
```

This is confusing, because it explicitly mentions JFIF. However, the referenced standard (ISO/IEC 10918-1, ITU-T T.81) does not specify JFIF but baseline JPEG, which allows compression in RGB color space.

Unfortunately, the API does not provide a way to specify the color space transform. Some of the methods, such as `tobytes()` have the `mode` parameter. However, this parameter only applies to  transformations upon decompression. The following hack provides a workaround:

```python
image = Image.open(BytesIO(jpeg_bitstream))

color_mode = 'YCbCr'
image.tile = [(
    'jpeg',
    image.tile[0][1],
    image.tile[0][2],
    (color_mode, '')
)]
image.mode = color_mode
image.rawmode = color_mode

uncompressed_bitstream = image.tobytes()
``` 

By setting `mode` to `"YCbCr"` (setting `mode` alone doesn't work, but `tile` and `rawmode` have to be updated as well), no transformation is applied upon decompression and images are decoded correctly.

Since JPEG may not provide any color information in the bitstream (in the absence of the APP14 marker), it would be great to provide a way for users to specify the color transform upon construction of the `Image` object.

**Comment by homm** [comment]
> Since JPEG may not provide any color information in the bitstream (in the absence of the APP14 marker), it would be great to provide a way for users to specify the color transform upon construction of the Image object.

In general, this is not what a processing library should do. An image should contain all necessary information for loading, that is why image formats are invented. «JPEG does not provide any color information» actually means color information should be assumed from channels number (three channels are always YCbCr, four channels are always CMYK).

What we actually can do is add support for the APP14 marker to extract color information from it. For this, **please provide some example image** that would be encoded in raw RGB with the correct APP14 marker.

If your application uses jpegs with non-standard color interpretation and you really need to change image mode without converting colors, you can use `Image.Image.split()` and `Image.merge` methods as shown in the example:

```python
im = Image.open('../imgs/dices.png')
Image.merge('CMYK', im.split()).save('../imgs/dices.jpeg', subsampling=0)

Image.merge('RGBA', Image.open('../imgs/dices.jpeg').split()).save('../imgs/dices.jpeg.png')
```

**Comment by hackermd** [comment]
> What we actually can do is add support for the APP14 marker to extract color information from it. For this, please provide some example image that would be encoded in raw RGB with the correct APP14 marker.

Thanks @homm for your feedback. I will generate a couple of example images for testing purposes.

**Comment by radarhere** [comment]
@hackermd #3953 would have resolved this, but you closed it in favour of #5408.

#5408 has now been merged - so does that mean that this is resolved?

**Comment by hackermd** [comment]
@radarhere this is a related but different problem. #5498 will enable the library to apply the correct color transform if the APP14 marker is present. If the marker is absent, however, there is not really a way for the library to know in what color space the image was compressed and hence what color transform would need to be applied. At the moment, the library just assumes that the pixel data represent a JPEG file and consequently that the image was transformed into YCbCr color space prior to compression according to the JPEG File Interchange Format (JFIF). However, the JPEG Interchange Format (JIF) is more general and there are situations where a JPEG image may not represent a JPEG file, but is included in a container format such as TIFF or DICOM. In that case, pixel data may not have been color transformed prior to compression and the color transformation should not be applied after decompression.

I don't think that there is a solution to this problem and am fine with resolving this issue. The only suggestion that I have is clarify the behavior in the [documentation](https://pillow.readthedocs.io/en/latest/handbook/image-file-formats.html#jpeg).
